labextension/src/serverproxy.ts

```diff
@@ -0,0 +1,90 @@
+import { JSONObject } from "@lumino/coreutils";
+import { ISignal } from "@lumino/signaling";
+import { ServerConnection } from "@jupyterlab/services";
+
+/**
+ * The server model for a proxy.
+ */
+export interface IModel extends JSONObject {
+  /**
+   * The name of the proxy app.
+   */
+  readonly name: string;
+
+  /**
+   * The cmd used to launch proxy app.
+   */
+  readonly cmd: string;
+
+  /**
+   * The port at which proxy app is running. Port 0 means unix socket.
+   */
+  readonly port: string;
+
+  /**
+   * The url endpoint of the proxy app.
+   */
+  readonly url: string;
+
+  /**
+   * Proxy app managed by jupyter-server-proxy or not.
+   */
+  managed: boolean;
+}
+
+/**
+ * The interface for a server proxy manager.
+ *
+ * The manager is responsible for maintaining the state of running
+ * server proxy apps.
+ */
+export interface IManager {
```

Comment:
This can `extend` other interfaces so one needn't retype e.g. `shutdownAll`.

Reply:
`ServerProxyManager` is now extended from `BaseManager` in `@jupyterlab/services`. Ref [commit](https://github.com/jupyterhub/jupyter-server-proxy/pull/395/commits/f25996b57d452ecfdd85fb5bf02b0f6c6ef147ae)